snuba/datasets/dataset_schemas.py

```diff
@@ -0,0 +1,44 @@
+from typing import Optional, List, Iterator
+
+from snuba.datasets.schema import Schema
+
+
+class DatasetSchemas(object):
+    """
+    A collection of schemas associated with a dataset, providing access to schemas and aggregated functions
```

Comment:
I would replace "aggregated functions" with DDL functions so you are defining a bit the responsibility of the class instead of being a bag for anything related with schemas.